crates/solidity/Cargo.toml

```diff
@@ -40,7 +40,7 @@ inkwell = { workspace = true }
 
 revive-common = { workspace = true }
 revive-llvm-context = { workspace = true }
-
+tracing= "0.1.41"
```

Comment:
Why is this added?

Use `log` instead given it is only being used for logging which `log` already provides, given `log` is already a workspace dependency. Please use existing workspace dependencies wherever possible.

New dependencies should be added to the workspace dependencies but adding this is unnecessary.

Reply:
The test will require modification of `standard_json` in lib.rs to be modified to not read from std hence i didnt include it here is that still fine?

Reply:
Logs are just for testing i wasnt sure what the rules were around logging but i will remove them.

Reply:
>  The test will require modification of standard_json in lib.rs to be modified to not read from std

Why?

> Logs are just for testing i wasnt sure what the rules were around logging but i will remove them.

Thanks

Reply:
@xermicus i wasnt able to get reading and writing to std in a test environment with out getting the BROKEN PIPE error or the test hanging since standard_json calls process.exit(0) which meant that reading the results from compilation will never execute

Reply:
Heres a code share of the file i used locally to test
https://codeshare.io/VN0Pl3 ignore any unused imports i was trying different solutions to get reading and writting to std but failed so had to take the easy route of modifying standard_json

Reply:
test file: can be found on this codeshare ensure the file is in the root of the solidity crate https://codeshare.io/DANvm3 @xermicus

Reply:
Alternatively i can push the local branch i created for testing then we can delete it after you verify if everything is ok?

Reply:
Additionally i used a source file that is similar to what foundry generates

Reply:
It should be testable in a [CLI test](https://github.com/paritytech/revive/tree/main/crates/solidity/src/tests/cli-tests).

Reply:
Aha i didnt focus on this part of the code do i need to add a test there?

Reply:
Yes. In the CLI tests you have complete control over the compiler invocation and configuration.

Reply:
Alright i wasnt aware i needed to change that part of the code will do so asap